internal/ad/krb5_test.go

```diff
@@ -0,0 +1,136 @@
+package ad_test
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/adsys/internal/ad"
+	"github.com/ubuntu/adsys/internal/testutils"
+)
+
+func TestTicketPath(t *testing.T) {
+	// Build mock libkrb5
+	var mockLibPath string
+	if os.Getenv("GO_WANT_HELPER_PROCESS") != "1" {
+		mockLibPath = testutils.BuildKrb5Mock(t, ".")
+	}
+
+	// We setup and rerun in a subprocess because we need to preload the mock libkrb5
+	if setupSubprocessForTest(t, mockLibPath) {
+		return
+	}
+
+	tests := map[string]struct {
+		krb5Behavior string
+		createCCache bool
+		ccacheIsDir  bool
+
+		wantErr bool
+	}{
+		"Lookup is successful":                 {krb5Behavior: "return_ccache:FILE:%s", createCCache: true},
+		"Allow ccache without FILE identifier": {krb5Behavior: "return_ccache:%s", createCCache: true},
+
+		"Error when ccache not present on disk": {krb5Behavior: "return_ccache:FILE:%s", wantErr: true},
+		"Error when ccache is a directory":      {krb5Behavior: "return_ccache:%s", createCCache: true, ccacheIsDir: true, wantErr: true},
+		"Error when initializing context":       {krb5Behavior: "error_initializing_context", wantErr: true},
```

Comment:
here, we don’t know if the error is due to `createCCache` not being true or the `krb5Behavior` (even if ok, the code is influenced by the krb5Behavior before the cache existence check). Shouldn’t we have a default to create a ccache (and so, we remove this repeated parameter everywhere) and then, have a noCreateCCache for the specific error case?